array_api_compat/common/_helpers.py

```diff
@@ -255,6 +254,166 @@ def is_array_api_obj(x):
         or is_pydata_sparse_array(x) \
         or hasattr(x, '__array_namespace__')
 
+def _compat_module_name():
+    assert __name__.endswith('.common._helpers')
+    return __name__.removesuffix('.common._helpers')
+
+def is_numpy_namespace(xp) -> bool:
+    """
+    Returns True if `xp` is a NumPy namespace.
+
+    This includes both NumPy itself and the version wrapped by array-api-compat.
+
+    See Also
+    --------
+
+    array_namespace
+    is_cupy_namespace
+    is_torch_namespace
+    is_ndonnx_namespace
+    is_dask_namespace
+    is_jax_namespace
+    is_pydata_sparse_namespace
+    is_array_api_strict_namespace
+    """
+    return xp.__name__ in {'numpy', _compat_module_name + '.numpy'}
```

Comment:
What happens when ``xp` is `numpy.array_api` from previous numpy releases? If it is wrapped I assume it will return true, but unsure for unwrapped `numpy.array_api`. (same comment applies to `cupy.array_api`)

Reply:
Indeed I don't think this will be captured - how is scikit-learn using `numpy.array_api`? In SciPy we just use array-api-strict now.

Reply:
I was thinking about users who use numpy<2 with scikit-learn. I don't know what happens then, my guess is they end up being wrapped? I thought that it would be as easy as adding `numpy.array_api` to the set to catch it, so didn't go and investigate.

So at least 50% of my question is "for my own education"

Reply:
> I was thinking about users who use numpy<2 with scikit-learn. I don't know what happens then, my guess is they end up being wrapped?

Users should not be passing arrays from `numpy.array_api` - a numpy<2 user just needs to pass a normal NumPy array, and `array_api_compat.numpy` handles the wrapping. `numpy.array_api` was purely for testing purposes, and that is array-api-strict now.

Reply:
Probably fine to skip it then? "mark as resolved"?

Reply:
numpy.array_api would never be wrapped. It's not really appropriate to consider it a "numpy" namespace because even though it lived in numpy it had a separate array object (this confusion is one reason it was split out).